### PR TITLE
Fix issues reported by JET

### PR DESCRIPTION
**Pull request by lgoettgens**
Resolves https://github.com/QuantumSavory/QuantumClifford.jl/issues/307.

- [x] The code is properly formatted and commented.
- [x] Substantial new functionality is documented within the docs.
- [x] All new functionality is tested.
- [x] All of the automated tests on github pass.
- [x] We recently started enforcing formatting checks. If formatting issues are reported in the new code you have written, please correct them.

### CODE REVIEW

File: src/mul_leftright.jl [hunk 1]
```diff
@@ -56,11 +56,14 @@ function mul_ordered_lv!(r::AbstractVector{T}, l::AbstractVector{T}; phases::Val
...
 =#
 
+# This method is already covered by the identical but less specific one right below. (And JET does not like it)
+#=
 function mul_ordered!(r::SubArray{T,1,P,I1,L1}, l::SubArray{T,1,P,I2,L2}; phases::Val{B}=Val(true)) where {T<:Unsigned, B, I1, I2, L1, L2, P<:Adjoint}
     # This method exists because SIMD.jl does not play well with Adjoint
     # Delete it and try `QuantumClifford.mul_left!(fastcolumn(random_stabilizer(194)), 2, 1)` # works fine for 192
     _mul_ordered_nonvec!(r,l; phases=B)
 end
+=#
```
Comment (lgoettgens) [inline src/mul_leftright.jl:66]: @Krastanov should I keep this method commented out or straight up remove it?

**Comment by lgoettgens** [comment]
This is good to go from my POV @Krastanov

**Comment by Krastanov** [comment]
This looks great, thank you! It will probably take me a couple of days to get around to completely reviewing it. I am curious about one particular point:

I see that in a lot of places you have changed old historic code like `x.rank` to much neater and idiomatic `rank(x)`. That is great, but I am curious why did that have an effect on the amount of JET warnings. Do you have a brief explanation?

**Comment by lgoettgens** [comment]
> I see that in a lot of places you have changed old historic code like `x.rank` to much neater and idiomatic `rank(x)`. That is great, but I am curious why did that have an effect on the amount of JET warnings. Do you have a brief explanation?

In an in-between version while fiddling around with some reports, I thought that one JET report was due to badly inferred `.tab` (or `.rank`, don't remember it exactly). My "solution" was to add a type assertion to a `tab(::Dunno)` return value, and update the call-sites to use this getter instead of direct field access. It turned out that the JET issue was resolved by some other change on its own, but I thought I'll just keep this improvement in the PR.

File: src/mul_leftright.jl (hunk 1 above)
Comment (Krastanov) [reply src/mul_leftright.jl]: yup, indeed. I just removed it while reviewing

**Comment by Krastanov** [comment]
Thank you, this is very much appreciated! This should be merged soon, just waiting on a final round of tests. I submitted the final list of bounties to NumFOCUS this week, and I will try to add this one as well, but there is a chance they will have to process it in the next round.

**Comment by Krastanov** [comment]
I merged master in here because there was a bug in LinearAlgebra 1.11.1 that we were hitting. The master branch has the workaround.

**Comment by lgoettgens** [comment]
> Thank you, this is very much appreciated! This should be merged soon, just waiting on a final round of tests. I submitted the final list of bounties to NumFOCUS this week, and I will try to add this one as well, but there is a chance they will have to process it in the next round.

Thanks!

**Pull request merged**